### ISSUE TITLE
Node overwrites local image even if it's newer

### ISSUE THREAD

**Issue by frankcorneliusmartin** [IKNL/vantage6-node#2]
When developing algorithms, local images may be newer than the those in the docker registry. Currently, the node pulls required images from the registry even when a newer image is already available. This overwrites the newer image with the old one.

Not sure if there's any way to prevent this, though?

**Comment by frankcorneliusmartin** [comment]
@mellesies It seems default behavior of Docker to pull the latest image from the repro (even if you have a local newer image) when you use the command `docker pull`.  

https://github.com/moby/moby/issues/10291
https://github.com/moby/moby/issues/14382

If you use `docker run` it only pulls an image if there is no local image present.

We have two options to solve this. 

1) We keep using a different tag locally (as suggested by the Moby community)
2) We do a lot of work and make sure the latest timestamp is kept
3) (my prefered option) We add an extra option to `vnode start` and `vserver start` to pull the latest image. If you do not specify `--update` no images will be pulled.

**Comment by mellesies** [comment]
This issue goes beyond `vnode` and `vserver`: it also applies to images with algorithms downloaded by the node.

For node/server I'm quite OK with your solution. For the algorithms, I'm not sure. Checking timestamps would be nice there, although using different tags might help preventing security issues.

**Comment by frankcorneliusmartin** [comment]
What if we use the convention a different tag that has priority (e.g. `local` or something similar)?

**Comment by mellesies** [comment]
Since we're using harbor, getting the creation date for an image is not too difficult. For example, to get the date for `vantage/vtg.chisq`:
`curl -i -k -X GET "https://harbor.distributedlearning.ai/api/repositories/vantage/vtg.chisq/tags/latest"`

Which returns:
```json
{
  "digest": "sha256:4e4a67b30e05c68d9a852dbe774db7c42d2c671694505257ad53c305591e5baf",
  "name": "test",
  "size": 402100892,
  "architecture": "amd64",
  "os": "linux",
  "os.version": "",
  "docker_version": "19.03.5",
  "author": "Melle Sieswerda \u003cm.sieswerda@iknl.nl\u003e",
  "created": "2020-02-12T16:45:11.107154Z",
  "config": {
    "labels": {
      "maintainer": "Melle Sieswerda \u003cm.sieswerda@iknl.nl\u003e",
      "org.label-schema.license": "GPL-2.0",
      "org.label-schema.vcs-url": "https://github.com/rocker-org/r-base",
      "org.label-schema.vendor": "Rocker Project"
    }
  },
  "signature": null,
  "labels": []
}
```

We could use that to check if the local image is newer.

**Comment by frankcorneliusmartin** [comment]
Implemented and merged in 1.1.0
